### PR TITLE
Best Practices for Reproducing Anaconda Environments

### LINKED ISSUE #567: Anaconda best practices for environment reproducibility
### What would you like to see added?

Blog post describing good practice for mixing conda and pip: https://www.anaconda.com/blog/using-pip-in-a-conda-environment

Also relevant: https://pythonspeed.com/articles/conda-dependency-management/

### PR DESCRIPTION
**Pull request by iam4tune**
# Pull Request

This is a pull request that provides information on Best Practices for reproducing Anaconda Environments.

## Overview

Included a section that details some of the best practices, researchers may need to follow when working with Anaconda Environments. Particularly as they collaborate and share their scripts, and projects with other researchers. Some of the content in the article, provide details around reproducibility, portability, etc.

## Proposed Changes

Included a section to using_anaconda.md

Fixes #567

### CODE REVIEW

**Review by wwarriner — CHANGES_REQUESTED** [review]
Overall this looks good! I have a few thoughts I've commented on in the changes. Happy to discuss.

File: docs/workflow_solutions/using_anaconda.md [hunk 1]
```diff
@@ -18,6 +18,26 @@ Anaconda can also install Pip and record which Pip packages are installed, so An
     If using Anaconda on Cheaha, please see our [Anaconda on Cheaha page](../cheaha/software/software.md#anaconda-on-cheaha) for important details and restrictions.
 <!-- markdownlint-enable MD046 -->
 
+## Anaconda best practices for environment reproducibility
+
+Creating, activating and managing environments with Anaconda can be challenging, especially when aiming for reproducibility across different systems, platforms, and projects. To tackle these challenges, below is a list of consolidated best practices drawn from some practical experiences:
+
+1. Using Conda and Pip Wisely: While using Cheaha, or any other platform prioritise using Conda packages as comprehensive compatibility and dependency checks are done during installation. Such that during project implementation, the required libraries function properly and would not throw up compatibility and dependency errors. Use Conda packages instead of Pip when possible as they significantly offer a reduced risk of breaking environments. If Pip packages are necessary for your work, install them after all Conda packages to minimize conflicts and breakage of your environment.
```
Comment (iam4tune) [inline docs/workflow_solutions/using_anaconda.md:25]: Include a sentence to state, how packages are installed in order from the yml file.

File: docs/workflow_solutions/using_anaconda.md [hunk 2]
```diff
@@ -18,6 +18,26 @@ Anaconda can also install Pip and record which Pip packages are installed, so An
...
+
+4. Export and Store Requirements: Keep a record of Conda and Pip requirements in text files. This practice enables easy sharing and version control, making it simpler to recreate environments on different machines.
+
+5. Consider Reproducibility and Upgradability: Balancing the need for reproducible builds with the flexibility to upgrade dependencies is crucial. Utilize tools like conda-lock to generate lock files that pin dependencies, allowing for reproducible environments that can be easily updated when necessary. Please note that using `conda-lock` will pin dependencies for reproducibility, ensuring exact version upgrades are done when needed.
+
+6. Address Portability Issues: Make your environment files as portable as possible by excluding system-specific paths and using general package versions where appropriate. This consideration ensures that environments can be replicated across different systems without modification.
+
+7. Utilize Third-Party Tools: Tools such as conda-lock can help manage dependencies more efficiently by creating lock files that facilitate fast and reproducible environment setups across different operating systems.
```
[the remark this answers is not in the document]
Comment (iam4tune) [reply docs/workflow_solutions/using_anaconda.md:37]: I would look this up, and create an article for conda-lock.

[the remark this answers is not in the document]
Comment (wwarriner) [reply docs/workflow_solutions/using_anaconda.md]: Sounds good, make sure to create a new issue for it that you can resolve later, as a reminder.

**Review by wwarriner — CHANGES_REQUESTED** [review]

**Review by wwarriner — APPROVED** [review]
Looks good!

**Pull request merged**